openex-api/src/main/java/io/openex/service/MailingService.java

```diff
@@ -19,53 +20,61 @@
+import static io.openex.config.SessionHelper.currentUser;
+
 
 @Service
 public class MailingService {
 
-    @Resource
-    protected ObjectMapper mapper;
+  @Resource
+  protected ObjectMapper mapper;
 
-    @Resource
-    private OpenExConfig openExConfig;
+  @Resource
+  private OpenExConfig openExConfig;
 
-    private ContractService contractService;
+  private ContractService contractService;
 
-    private ApplicationContext context;
+  private ApplicationContext context;
 
-    @Autowired
-    public void setContext(ApplicationContext context) {
-        this.context = context;
-    }
+  private UserRepository userRepository;
 
-    @Autowired
-    public void setContractService(ContractService contractService) {
-        this.contractService = contractService;
-    }
+  @Autowired
+  public void setUserRepository(UserRepository userRepository) {
+    this.userRepository = userRepository;
+  }
 
-    public void sendEmail(String subject, String body, List<User> users, Optional<Exercise> exercise) {
-        EmailContent emailContent = new EmailContent();
-        emailContent.setSubject(subject);
-        emailContent.setBody(body);
-        Inject inject = new Inject();
-        inject.setContent(mapper.valueToTree(emailContent));
-        inject.setContract(EmailContract.EMAIL_DEFAULT);
-        inject.setUser(currentUser());
-        Contract contract = contractService.resolveContract(inject);
-        if (contract == null) {
-            throw new UnsupportedOperationException("Unknown inject contract " + inject.getContract());
-        }
-        inject.setType(contract.getConfig().getType());
-        exercise.ifPresent(inject::setExercise);
-        List<ExecutionContext> userInjectContexts = users.stream().distinct()
-                .map(user -> new ExecutionContext(openExConfig, user, inject, "Direct execution")).toList();
-        ExecutableInject injection = new ExecutableInject(false, true, inject, contract, List.of(), userInjectContexts);
-        Injector executor = context.getBean(contract.getConfig().getType(), Injector.class);
-        executor.executeInjection(injection);
-    }
+  @Autowired
+  public void setContext(ApplicationContext context) {
+    this.context = context;
+  }
+
+  @Autowired
+  public void setContractService(ContractService contractService) {
+    this.contractService = contractService;
+  }
 
-    public void sendEmail(String subject, String body, List<User> users) {
-        sendEmail(subject, body, users, Optional.empty());
+  public void sendEmail(String subject, String body, List<User> users, Optional<Exercise> exercise) {
+    EmailContent emailContent = new EmailContent();
+    emailContent.setSubject(subject);
+    emailContent.setBody(body);
+    Inject inject = new Inject();
+    inject.setContent(mapper.valueToTree(emailContent));
+    inject.setContract(EmailContract.EMAIL_DEFAULT);
+    inject.setUser(userRepository.findById(currentUser().getId()).orElseThrow());
```

Comment:
Can you add the keyword `this.` when you are using class's properties ?
It's more readable.